### PR TITLE
Partial Main Loop Rewrite ARMInfo and Process Handler with Unittesting

### LINKED ISSUE #589: Main Loop Logic Rewrite and QoL Upgrades
### **Is your feature request related to a problem? Please describe.**
Currently, the main loop execution path is decided by a matrixed truth table of options that tries to account for each combination of settings before executing a rip. This has gotten to the point where modifying core logic, or even predicting it's behavior, is unacceptably complex. To address this, I will be rewriting the core logic in of ARM, implementing the [Stranger Fig Pattern](https://learn.microsoft.com/en-us/azure/architecture/patterns/strangler-fig) to make it available for release in piecemeal fashion.

### **Describe the solution you'd like**
Currently, the logic doesn't know what type of video disk it's working with when determining the steps to go through. This compounds the multiple layers of checks it needs to make, leading to code that is hard to read, debug, and maintain. This rewrite will break each type out into it's own execution path, while maintaining the minimal amount of code duplication that we've already achieved. 

This issue aims to track the pull requests that will cover fixing this problem, of which there should be at least 6:
- [x] 1. Separate RIPMETHOD variables in `arm.yaml` for DVD, Blu-ray (#590)
- [ ] 2. Migrate DVD ripping to new architecture
- [ ] 3. Migrate Blu-ray ripping to new architecture
- [ ] 4. Migrate music ripping to new architecture
- [ ] 5. Migrate dataripping to new architecture
- [ ] 6. Excise deprecated logic and variables

Additionally, this rewrite aims to achieve 2 further goals across each step.
1. Ensure that each job is executed as a thread separate from the UI to prevent hangs on the page
2. Minimize the amount of time the database is being accessed as much as possible

### **Are you able to help code/implement this feature ?**
Yes. I will take full responsibility for the development of this

### PR DESCRIPTION
**Pull request by microtechno9000**
# Description
This commit makes some changes to the ARM main loop for ripping media
Changes:
1. ARM version information (git, database, alembic, python) moved into a separate class ARMInfo for easier management and testing
2. Unittest created for ARMInfo
3. python sub-process handler created and used by ARMInfo to allow for easier management and testing
4. Unittest created for the ARM process handler

Does not change the function of the ripper code, but has broken out portions of the code to allow for easier management and testing

Fixes #589 
Part of the re-write project

## Type of change
Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.
Please also list any relevant details for your test configuration

- [x] Docker
- [x] Ubuntu 20.04.6 LTS

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have tested that my fix is effective or that my feature works

# Changelog:
as above in description

# Logs
Music log extract, first few lines (docker)
```
[06-19-2023 11:51:52] INFO ARM: ARM version: 2.6.40
[06-19-2023 11:51:52] INFO ARM: Python version: 3.8.10 (default, Mar 13 2023, 10:26:41) 
[GCC 9.4.0]
[06-19-2023 11:51:52] INFO ARM: User is: arm
[06-19-2023 11:51:52] INFO ARM: Alembic head is: 2e0dc31fcb2e
[06-19-2023 11:51:52] INFO ARM: Database version is: 2e0dc31fcb2e
[06-19-2023 11:51:52] INFO ARM: ************* Starting ARM processing at 2023-06-19 11:51:52.587728 *************
```

Music log extract, first few lines. (ubuntu)
```
[06-19-2023 21:30:48] INFO ARM: ARMInfo.get_values ARM version: 2.6.40
[06-19-2023 21:30:48] INFO ARM: ARMInfo.get_values Python version: 3.8.10 (default, May 26 2023, 14:05:08) 
[GCC 9.4.0]
[06-19-2023 21:30:48] INFO ARM: ARMInfo.get_values User is: mtech
[06-19-2023 21:30:48] INFO ARM: ARMInfo.get_values Alembic head is: 2e0dc31fcb2e
[06-19-2023 21:30:48] INFO ARM: ARMInfo.get_values Database version is: 2e0dc31fcb2e
[06-19-2023 21:30:48] INFO ARM: main.<module> ************* Starting ARM processing at 2023-06-19 21:30:48.118625 *************
```

Unit testing output from `./devtools -pr` attached below
[devtools_pr.txt](https://github.com/automatic-ripping-machine/automatic-ripping-machine/files/11787903/devtools_pr.txt)

### CODE REVIEW

File: scripts/thickclient/arm_wrapper.sh [hunk 1]
```diff
@@ -4,7 +4,7 @@ set -eo pipefail
 
 DEVNAME=$1
 PROTECTION=""
-USER="arm"
+USER="mtech"
```
Comment (shitwolfymakes) [inline scripts/thickclient/arm_wrapper.sh:7]: Please revert this change

**Review by shitwolfymakes — CHANGES_REQUESTED** [review]
Make this change and please attach the log of a successfully completed run

File: scripts/thickclient/arm_wrapper.sh (hunk 1 above)
Comment (microtechno9000) [reply scripts/thickclient/arm_wrapper.sh:7]: again. slipped through to the keeper
will fix

**Comment by microtechno9000** [comment]
resolved, will do a run tonight and attach the logs

**Comment by microtechno9000** [comment]
Completed, was only info level logging, but no issues were raised

[Parkway-Drive-Darker-Still.log](https://github.com/automatic-ripping-machine/automatic-ripping-machine/files/11833358/Parkway-Drive-Darker-Still.log)

**Review by shitwolfymakes — APPROVED** [review]
